pkgs/development/libraries/librsvg/default.nix

```diff
@@ -16,7 +16,10 @@ stdenv.mkDerivation rec {
 
   outputs = [ "out" "dev" "installedTests" ];
 
-  buildInputs = [ libxml2 bzip2 pango libintl ];
+  buildInputs = [ libxml2 bzip2 pango libintl ]
+    ++ lib.optionals stdenv.isDarwin [ darwin.libobjc ];
+
+  NIX_LDFLAGS = if stdenv.isDarwin then "-lobjc" else null;
```

Comment:
A bit hacky / confusing but rewriting this as

```
  ${if stdenv.hostPlatform.isDarwin then "NIX_LDFLAGS" else null } = "-lobjc";
```

should avoid a Linux rebuild for now if that's a concern here.

Reply:
Hehe, that's neat, didn't actually know it was possible to circumvent the problem this way (though that should ideally be the meaning of `foo = null` tbh.). 

If we want to target master then I think we do want this hack, preferably with a `FIXME` comment.

Reply:
I'm inclined to the initial version, because is more easy to understand; but I don't mind updating it.  Please let me know if we are going to use this approach

Reply:
Yours' definitely much clearer so I would also be in favour of that. I think I may have left my comment at the time there was still a change triggering a Linux rebuild, as that's no longer an issue so you can just dismiss my comment.

Reply:
Right, I guess `optionalString` returns an empty string when false (which does make sense), but `foo = null` actually works the way we want. That makes it a lot more palatable indeed :)